tests/unit/test_model_creator.py

```diff
@@ -1,155 +1,204 @@
         accel="ps",
-        nat_tunes=[6.32, 6.29],
+        nat_tunes=[0.21, 0.323], # from madx_job file in acc_models
         dpp=0.0,
         energy=1.4,
-        year=2018,
-        modifiers=[PS_MODEL / "2018" / "strength" / "PS_LE_LHC_low_chroma.str"],
+        year="2018",
+        fetch=PATHFETCHER,
+        path=acc_models_ps_2021,
+        scenario="lhc",
+        cycle_point="2_flat_top",
+        str_file="ps_ft_lhc.str",
+        tune_method="pfw",
     )
     accel = create_instance_and_model(
         type="nominal", outputdir=tmp_path, logfile=tmp_path / "madx_log.txt", **accel_opt
     )
     check_accel_from_dir_vs_options(tmp_path, accel_opt, accel, required_keys=["year"])
 
+    accel_opt_duplicate = accel_opt.copy()
+    accel_opt_duplicate["energy"] = None
 
-@pytest.mark.basic
-def test_ps_creation_nominal_free_2021(tmp_path):
-    accel_opt = dict(
-        accel="ps",
-        nat_tunes=[6.32, 6.29],
-        dpp=0.0,
-        energy=1.4,
-        year=2021,
-        modifiers=[PS_MODEL / "2021" / "strength" / "ps_fb_lhc.str"],
-    )
-    accel = create_instance_and_model(
-        type="nominal", outputdir=tmp_path, logfile=tmp_path / "madx_log.txt", **accel_opt
-    )
-    check_accel_from_dir_vs_options(tmp_path, accel_opt, accel, required_keys=["year"])
+    with pytest.raises(RuntimeError):
+        create_instance_and_model(
+            type="nominal", outputdir=tmp_path, logfile=tmp_path / "madx_log.txt", **accel_opt_duplicate
+        )
 
 
 @pytest.mark.basic
-def test_lhc_creation_nominal_driven(tmp_path):
+def test_lhc_creation_nominal_driven(tmp_path, acc_models_lhc_2023):
     accel_opt = dict(
         accel="lhc",
-        year="2018",
-        ats=True,
+        year="2023",
         beam=1,
         nat_tunes=[0.31, 0.32],
         drv_tunes=[0.298, 0.335],
         driven_excitation="acd",
         dpp=0.0,
-        energy=6.5,
-        modifiers=[COMP_MODEL / "opticsfile.24_ctpps2"],
+        energy=6800.0,
+        fetch=PATHFETCHER,
+        path=acc_models_lhc_2023,
+        modifiers=LHC_30CM_MODIFIERS,
     )
     accel = create_instance_and_model(
         outputdir=tmp_path, type="nominal", logfile=tmp_path / "madx_log.txt", **accel_opt
     )
     check_accel_from_dir_vs_options(tmp_path, accel_opt, accel, required_keys=["beam", "year"])
 
+    # checks that should fail
+
+    with pytest.raises(AcceleratorDefinitionError):
+        accel_duplicate = copy.deepcopy(accel)
+        accel_duplicate.model_dir = None
+        LhcModelCreator.check_accelerator_instance(accel_duplicate)
+
+    with pytest.raises(AcceleratorDefinitionError):
+        accel_duplicate = copy.deepcopy(accel)
+        accel_duplicate.modifiers = None
+        LhcModelCreator.check_accelerator_instance(accel_duplicate)
+
+    with pytest.raises(AttributeError):
+        create_instance_and_model(
+            type="nominal", outputdir=None, logfile=tmp_path / "madx_log.txt", **accel_opt
+        )
+
```

Comment:
For all these `pytest.raises` it would be nice if we saved the exception info and checked for instance in the error message that it is what is expected, and not that the test passes because something unrelated / unexpected raises.

This would read as:
```python
with pytest.raises(AcceleratorDefinitionError) as excinfo:
    ... # stuff that should raise
    assert "expected error" in str(excinfo.value)  # to adapt of course
```
for instance

Reply:
The `assert` needs to be outside of the `with` statement, though:

```python
with pytest.raises(AcceleratorDefinitionError) as excinfo:
    ... # stuff that should raise
assert "expected error" in str(excinfo.value)  # to adapt of course
```

Reply:
Yes, good catch

Reply:
I thought about this, but that's very flaky. Imagine someone changes the message. A distinct Error Type would be the ideal solution, but I am too lazy for that

Reply:
> The `assert` needs to be outside of the `with` statement, though:
> 
> ```python
> with pytest.raises(AcceleratorDefinitionError) as excinfo:
>     ... # stuff that should raise
> assert "expected error" in str(excinfo.value)  # to adapt of course
> ```
> 
> otherwise it will be "skipped" by the raising of the errror

Are you sure about that?

I mean, if that's the case, then so be it. But that makes no sens. first `assert` is not `raise`, and second it should only catch `AcceleratorDefinitionError`, and still fail on any other type of error.

and third, `excinfo` should not even be in scope outside of the `with`

Reply:
According to the [pytest documentation](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=pytest%20raises#pytest.raises) it has to be outside of the `with` context block.

Reply:
> I thought about this, but that's very flaky. Imagine someone changes the message, or some other message says "expected error" for some reason.
> 
> A distinct Error Type would be the ideal solution, but I am too lazy for that. Or some `Kind` enum inside `AcceleratorDefinitionError`:
> 
> ```python
> assert error.kind == AcceleratorDefinitionError.NoModelDir
> ```

It's a valid point, but I would personnally prefer having to deal with an update of the test if we ever change the error message, since this check would allow us to make sure the right code branch is raising and we're not lucking our way through the test.

Reply:
and ... check